app_flair.py

```diff
@@ -96,6 +123,103 @@ def parse_prediction_request(json_object: JsonDict) -> PredictionRequest:
     return PredictionRequest(layer, feature, projectId, Document(xmi, documentId, userId), typesystem)
 
 
+def dataset_split(list_sentences: list):
+    l = len(list_sentences)
+    test_size: int = round(l / 10)
+    dev_size: int = round(l / 10)
+    random.shuffle(list_sentences)
```

Comment:
I think you are shuffling the wrong list here

Reply:
@bjtulinl Did you have a look at this?

Reply:
Yes. But it seems that the function train_test_split in sklearn is used to split the dataset with the labels. In flair they use random_split in torch. So I write the function like this:
def dataset_split(list_sentences: list):
    l = len(list_sentences)
    dev_size: int = round(l / 10)
    train, dev = random_split(list_sentences,[l-dev_size,dev_size])
    train_l = len(train)
    test_size: int = round(train_l / 10)
    train, test = random_split(train, [train_l-test_size,test_size])
    return train, dev, test
Does it make sense?

Reply:
in sklearn you do not need to the labels. Your split is difficult to read, as you split from train twice, making it not a 80 10 10 split and checking the indices is quite hard.

Reply:
Then maybe:
def dataset_split(list_sentences: list):
    l = len(list_sentences)
    dev_size: int = round(l / 10)
    test_size: int = round(l / 10)
    train, dev = random_split(list_sentences,[l-dev_size,dev_size])
    train, test = random_split(train, [train_l-test_size,test_size])
    return train, dev, test
Or do you think sklearn is a better choice?

Reply:
Then maybe:
def dataset_split(list_sentences: list):
l = len(list_sentences)
dev_size: int = round(l / 10)
test_size: int = round(l / 10)
train, dev = random_split(list_sentences,[l-dev_size,dev_size])
train, test = random_split(train, [I-dev_size-test_size,test_size])
return train, dev, test
Or do you think sklearn is a better choice?

Reply:
I just would use sklearn. Do you really need the test set in flair?

Reply:
No. We can just set the corpus as:
    corpus = Corpus( train=train , dev=dev, test='')
"train" is used to train the model and "dev" is used to evaluate the score of the model. And the learning rate is only changed  when the f1 score of dev set decreases.

Reply:
Then I would just use either torch or sklearn split once and ignore the test set. That should make it simpler.